events/src/main/java/tech/pegasys/artemis/events/DirectEventDeliverer.java

```diff
@@ -13,24 +13,58 @@
 
 package tech.pegasys.artemis.events;
 
+import static tech.pegasys.artemis.metrics.ArtemisMetricCategory.EVENTBUS;
+
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.Counter;
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
 
 class DirectEventDeliverer<T> extends EventDeliverer<T> {
   private final ChannelExceptionHandler exceptionHandler;
+  private final LabelledMetric<Counter> consumedEventCounter;
+  private final LabelledMetric<Counter> failedEventCounter;
 
-  DirectEventDeliverer(final ChannelExceptionHandler exceptionHandler) {
+  DirectEventDeliverer(
+      final ChannelExceptionHandler exceptionHandler, final MetricsSystem metricsSystem) {
+    super(metricsSystem);
     this.exceptionHandler = exceptionHandler;
+    consumedEventCounter =
+        metricsSystem.createLabelledCounter(
+            EVENTBUS,
+            "event_consumed_count",
+            "Total number of events consumed",
+            "channel",
+            "subscriber");
+    failedEventCounter =
+        metricsSystem.createLabelledCounter(
+            EVENTBUS,
+            "event_failed_count",
+            "Number of events which failed to be processed",
+            "channel",
+            "subscriber");
   }
 
   @Override
   protected void deliverTo(final T subscriber, final Method method, final Object[] args) {
     try {
       method.invoke(subscriber, args);
     } catch (IllegalAccessException e) {
+      incrementCounter(failedEventCounter, subscriber, method);
       exceptionHandler.handleException(e, subscriber, method, args);
     } catch (InvocationTargetException e) {
+      incrementCounter(failedEventCounter, subscriber, method);
       exceptionHandler.handleException(e.getTargetException(), subscriber, method, args);
+    } finally {
+      incrementCounter(consumedEventCounter, subscriber, method);
     }
   }
+
+  private void incrementCounter(
+      final LabelledMetric<Counter> counter, final T subscriber, final Method method) {
+    counter
+        .labels(method.getDeclaringClass().getSimpleName(), subscriber.getClass().getSimpleName())
```

Comment:
Should we include `method.getName()`?

Reply:
Good question... I'm tempted to leave it out for now - every combination of label values introduces a whole new timeline to store so it would increase the storage required.  I suspect that most channels will only have a small number of highly related methods on them so we won't need that level of detail.  Previous experience with these kinds of events and monitoring lines up with that, but we won't know for sure until we use this to monitor actual nodes in production.